class.jetpack-jitm.php

```diff
@@ -56,490 +55,268 @@ function get_emblem()
+		}
+
+		if ( isset( $hide_jitm[ $feature_class ] ) ) {
+			if ( ! is_array( $hide_jitm[ $feature_class ] ) ) {
+				$hide_jitm[ $feature_class ] = array( 'last_dismissal' => 0, 'number' => 0 );
+			}
+		} else {
+			$hide_jitm[ $feature_class ] = array( 'last_dismissal' => 0, 'number' => 0 );
+		}
 
-		$upload_url   = add_query_arg( 'mode', 'grid', admin_url( 'upload.php' ) );
-		$new_post_url = admin_url( 'post-new.php' );
+		$number = $hide_jitm[ $feature_class ]['number'];
 
-		$msg = sprintf( __( 'Only videos uploaded from within the <a href="%s">media library</a> or while creating a <a href="%s">new post</a> will be fully hosted by WordPress.com.', 'jetpack' ), esc_url( $upload_url ), esc_url( $new_post_url ) );
-		?>
-        <div class="jp-jitm" data-track="videopress-upload-warning" data-stats_url="<?php echo esc_url( $jitm_stats_url ); ?>">
-            <!-- <a href="#" data-module="videopress" class="dismiss"><span class="genericon genericon-close"></span></a>-->
-
-			<?php echo self::get_emblem(); ?>
-
-            <p class="msg">
-				<?php echo $msg; ?>
-            </p>
-            <p>
-                <a href="<?php echo esc_url( $upload_url ); ?>" title="<?php esc_attr_e( 'Upload a Video', 'jetpack' ); ?>" data-module="videopress" data-jptracks-name="nudge_click" data-jptracks-prop="jitm-videopress-upload" class="button button-jetpack launch jptracks"><?php esc_html_e( 'Upload a Video Now', 'jetpack' ); ?></a>
-            </p>
-        </div>
-		<?php
+		$hide_jitm[ $feature_class ] = array( 'last_dismissal' => time(), 'number' => $number + 1 );
+
+		Jetpack_Options::update_option( 'hide_jitm', $hide_jitm );
+
+		return true;
 	}
 
 	/**
-	 * Display message prompting user to install WooCommerce Services.
+	 * Asks the wpcom API for the current message to display keyed on query string and message path
 	 *
-	 * @since 4.6
+	 * @param $message_path string The message path to ask for
+	 * @param $query string The query string originally from the front end
+	 *
+	 * @return array The JITM's to show, or an empty array if there is nothing to show
 	 */
-	function woocommerce_services_msg() {
-		if ( ! current_user_can( 'manage_woocommerce' ) || ! current_user_can( 'install_plugins' ) ) {
-			return;
+	static function get_messages( $message_path, $query ) {
+		// custom filters go here
+		add_filter( 'jitm_woocommerce_services_msg', array( 'Jetpack_JITM', 'jitm_woocommerce_services_msg' ) );
+		add_filter( 'jitm_jetpack_woo_services_install', array( 'Jetpack_JITM', 'jitm_jetpack_woo_services_install' ) );
+		add_filter( 'jitm_jetpack_woo_services_activate', array(
+			'Jetpack_JITM',
+			'jitm_jetpack_woo_services_activate'
+		) );
+
+		$user = wp_get_current_user();
+
+		// unauthenticated or invalid requests just bail
+		if ( ! $user ) {
+			return array();
 		}
 
-		if ( isset( self::$jetpack_hide_jitm['woocommerce_services'] ) ) {
-			return;
-		}
+		require_once( JETPACK__PLUGIN_DIR . 'class.jetpack-client.php' );
```

Comment:
This is not the case when loaded by the api endpoint, and needs to be manually required.